src/main/java/seedu/contax/model/appointment/Appointment.java

```diff
@@ -100,10 +132,16 @@ public boolean equals(Object other) {
         }
 
         Appointment otherAppointment = (Appointment) other;
+
+        Priority priority = (getPriority() != null) ? getPriority() : Priority.NONE;
+        Priority otherPriority = (otherAppointment.getPriority() != null)
+                ? otherAppointment.getPriority() : Priority.NONE;
```

Comment:
You can just follow how the persons check is done to handle null fields.

Reply:
Person model dont have null fields

Reply:
Person edit null using editDescriptor but Descriptor itself is still not null